src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -1474,37 +1476,53 @@ void Graph::Infer(SyncInferRequest* request) {
 void Graph::SortTopologically() {
     OV_ITT_SCOPE(FIRST_INFERENCE, itt::domains::intel_cpu_LT, "Graph::SortTopologically");
 
+    for (auto &node : graphNodes) {
+        // Sort in / out Parent edges by port index
+        // Make first N (N == port_num) edge indexes match with port index
+       std::sort(node->parentEdges.begin(), node->parentEdges.end(), [](const EdgeWeakPtr& lhs, const EdgeWeakPtr& rhs) {
+           auto lhs_ptr = lhs.lock();
+           auto rhs_ptr = rhs.lock();
+           OPENVINO_ASSERT(lhs_ptr && rhs_ptr);
+           return lhs_ptr->getOutputNum() < rhs_ptr->getOutputNum();
+        });
+
+        // Set execIndex of all nodes to default invaild value
+        node->execIndex = -1;
+    }
```

Comment:
Isn't it basically the same logiс which is done in line 1540, but for parent edges?
Also, the current assumption on master is that parent edges should be always sorted by output port number. It is done in scope of the addParentEdge method of the Node class. Could you please clarify, how it happened, that parent edges are not already sorted at this point? This could be the main root cause of the issue. I would assume that there is some transformation which goes around the graph / node api and inserts the edge using some custom logic.

What about implementing the following approach:
1) Keep a core part of the current graph traversal untouched
2) But to always start the traversal from the Input nodes, which are stored separately in scope of inputNodesMap, which is not affected by the SortTopologically method.

Would it resolve the issue?

Reply:
Yes, it is the same logic in line 1540. I guess there might be somewhere that make child edges are not in order, so that's why the code in line 1540 has to put in this function. So I write the similar codes for parent edges. Because the following sorting method depends on the condition that the parent edges has to sort according to the output port number. Actually I'm not sure whether the parent edges are always in order. I just write the code here to make sure it is.

Reply:
I didn't get what the main root cause of the issue here is refered to? And which issue does your approach mean to solve?

Reply:
This is the draft implementing the approach